### PR TITLE
Add support for CC 6.x double atomicAdd

### PR DESCRIPTION
**Pull request by BenjaminW3**

### CODE REVIEW

File: include/alpaka/exec/ExecGpuCudaRt.hpp [hunk 1]
```diff
@@ -114,8 +114,6 @@ namespace alpaka
 #if (!__GLIBCXX__) // libstdc++ even for gcc-4.9 does not support std::is_trivially_copyable.
             static_assert(
                 meta::Conjunction<
-                    // This true_ is required for the zero argument case because and_ requires at least two arguments.
```
Comment (psychocoderHPC) [inline include/alpaka/exec/ExecGpuCudaRt.hpp]: Is this change related to the atomicAdd support for sm_60?

Comment (BenjaminW3) [inline include/alpaka/exec/ExecGpuCudaRt.hpp]: Not really. It just removes some useless lines.

Comment (psychocoderHPC) [inline include/alpaka/exec/ExecGpuCudaRt.hpp]: Could you please move this change to a separate pull request. It removes a argument from a class and we should have a documentaion why we now allowed to remove it.

Comment (BenjaminW3) [inline include/alpaka/exec/ExecGpuCudaRt.hpp]: I will move it. It is not necessary because there is already at least one argument. The removal of the true_type has been forgotten long ago.
